src-electron/db/query-session.js

```diff
@@ -75,6 +75,27 @@ async function getSessionDirtyFlag(db, sessionId) {
     return dbApi.fromDbBool(row.DIRTY)
   }
 }
+/**
+ * Resolves with true or false, depending whether this session is dirty.
+ *
+ * @export
+ * @param {*} db
+ * @param {*} sessionId
+ * @returns A promise that resolves into true or false, reflecting session dirty state.
+ */
+async function getSessionUpgradeFlag(db, sessionId) {
+  let row = await dbApi.dbGet(db, 'SELECT UPGRADE FROM UPGRADE')
+  if (row.UPGRADE == 0) {
```

Comment:
I need to add another condition for null

Reply:
See above comment about parsing UPGRADE